Tasks/Gradle/CodeAnalysis/checkstyle.gradle

```diff
@@ -16,8 +16,12 @@ rootProject {
 
                 proj.apply plugin: 'checkstyle'
                 checkstyle {
-                    println(rootDir);
-                    configFile rootProject.file('checkstyle.xml')
+                    // Groovy way of resolving the current directory  
+                    File currentDir = file(new File("."));
+                    // Usually the rules xml sits with the project, but we don't want to touch the source files 
```

Comment:
Some more documentation in the source would be useful e.g. _We are using a hard-coded checkstyle.xml file that is checked in with as part of this task. We expect to find it in the same folder as this file. The user cannot specify their own checkstyle.xml file._